server/src/main/java/org/elasticsearch/persistent/PersistentTasksNodeService.java

```diff
@@ -183,12 +184,18 @@ public Task createTask(long id, String type, String action, TaskId parentTaskId,
                 task.markAsFailed(e);
             }
             processed = true;
+        }
+        catch (Exception e) {
+            initializationException = e;
         } finally {
             if (processed == false) {
                 // something went wrong - unregistering task
                 logger.warn("Persistent task [{}] with id [{}] and allocation id [{}] failed to create", task.getAction(),
                         task.getPersistentTaskId(), task.getAllocationId());
                 taskManager.unregister(task);
+                if (initializationException != null) {
+                    notifyMasterOfFailedTask(taskInProgress, initializationException);
```

Comment:
I wonder if we should use `task.markAsFailed` instead (which also unregisters) and if we can, then we can unify this into one catch block a few lines up? I think that would make this easier to read.

Reply:
I'm not sure if we should use `task.markAsFailed` if the task failed to initialize? it might be in an inconsistent state? That was the main reason for me to follow this approach. But we can do as you suggest.

Reply:
We could miss some cases where the `task.init` call the super class implementation after code that throws the exception, so we could miss some cases if we rely on `task.markAsFailed` as it uses the data provided in the parameters to notify the master node.